src/coreclr/vm/amd64/CrtHelpers.asm

```diff
@@ -36,7 +36,24 @@ LEAF_ENTRY JIT_MemSet, _TEXT
 
         cmp     byte ptr [rcx], 0       ; check dest for null
 
+ifdef HAS_ASAN
```

Comment:
Changes in this file are workarounds for https://github.com/dotnet/runtime/issues/80140